### PR TITLE
Story/118274 update zgw modules to use technical logging framework

### PR DESCRIPTION
**Pull request by Klaas-Ritense**

### CODE REVIEW

File: web/src/main/kotlin/com/ritense/valtimo/web/logging/CopyHeadClientHttpResponse.kt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ */
+
+package com.ritense.valtimo.web.logging
+
+import org.springframework.http.HttpHeaders
+import org.springframework.http.HttpStatusCode
+import org.springframework.http.client.ClientHttpResponse
+import java.io.InputStream
+
+class CopyHeadClientHttpResponse(
```
Comment (Tom-Ritense) [inline web/src/main/kotlin/com/ritense/valtimo/web/logging/CopyHeadClientHttpResponse.kt]: Copy naming could maybe be a little bit better, suggestions: CachedHead... , CopiedHead

File: web/src/main/kotlin/com/ritense/valtimo/web/logging/LoggingRestClientCustomizer.kt [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+import org.springframework.http.client.ClientHttpRequestInterceptor
+import org.springframework.http.client.ClientHttpResponse
+import org.springframework.stereotype.Component
+import org.springframework.web.client.HttpClientErrorException
+import org.springframework.web.client.RestClient
+import kotlin.text.Charsets.UTF_8
+
+@Component
+@SkipComponentScan
+class LoggingRestClientCustomizer : RestClientCustomizer, ClientHttpRequestInterceptor {
```
Comment (Tom-Ritense) [inline web/src/main/kotlin/com/ritense/valtimo/web/logging/LoggingRestClientCustomizer.kt:33]: Not super happy that both interfaces are used in a single class. Rather I would split them up. 1 RestClientCustomizer and 1 ClientHttpRequestInterceptor.

**Comment by ThomasMinkeRitense** [comment]
@Klaas-Ritense Shouldn't this go to feature/logging instead of next-minor?

File: web/src/main/kotlin/com/ritense/valtimo/web/logging/CopyHeadInputStream.kt [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+ * limitations under the License.
+ */
+
+package com.ritense.valtimo.web.logging
+
+import mu.KotlinLogging
+import java.io.IOException
+import java.io.InputStream
+
+class CopyHeadInputStream(
```
Comment (ivo-ritense) [inline web/src/main/kotlin/com/ritense/valtimo/web/logging/CopyHeadInputStream.kt]: I think we should write a test for this class.

Comment (Klaas-Ritense) [reply web/src/main/kotlin/com/ritense/valtimo/web/logging/CopyHeadInputStream.kt]: Added

File: web/src/main/kotlin/com/ritense/valtimo/web/logging/LoggingRestClientCustomizer.kt (hunk 2 above)
Comment (Klaas-Ritense) [reply web/src/main/kotlin/com/ritense/valtimo/web/logging/LoggingRestClientCustomizer.kt:33]: hmm I kinda like it this way tbh

File: web/src/main/kotlin/com/ritense/valtimo/web/logging/CopyHeadClientHttpResponse.kt (hunk 1 above)
Comment (Klaas-Ritense) [reply web/src/main/kotlin/com/ritense/valtimo/web/logging/CopyHeadClientHttpResponse.kt]: Changed

File: web/src/main/kotlin/com/ritense/valtimo/web/logging/LoggingRestClientCustomizer.kt (hunk 2 above)
Comment (Tom-Ritense) [reply web/src/main/kotlin/com/ritense/valtimo/web/logging/LoggingRestClientCustomizer.kt:33]: Alright

Comment (ivo-ritense) [reply web/src/main/kotlin/com/ritense/valtimo/web/logging/LoggingRestClientCustomizer.kt:33]: I think it's fine to combine them both in this case.

**Review by Tom-Ritense — APPROVED** [review]

**Review by ivo-ritense — APPROVED** [review]

**Pull request merged**